concepts/type-theory-glossary.md

```diff
@@ -403,6 +404,6 @@ The core features of IPLD Schemas map very closely to the categories of types we
 IPLD Schemas don't have a concept of [interfaces](#open-vs-closed-membership).
 At least, *not exactly* -- not *within* the schema.
 Instead, the *whole Schema* is treated as an interface:
-since you can take some data and take a schema, and attempt to unify them to get typed data if the schema "matches"...
+since you can take some data and take a schema, and attempt to unify them to get typed value if the schema "matches"...
```

Comment:
So sorry, I found one more when I read it again this morning.  I think this either needs to be "data" or "values" (plural).  I think I'd rather keep it "data" to agree with the earlier part of the sentence :)

Reply:
No worries :). I have changed it back to _data_.

Anyhow, working with Rust past few years, I found this nomenclature,

* _data_, one or more opaque blocks of bytes.
* _value_, data with its semantics understood.
* _type_, set of values belonging to same category.

feels more precise. If I understand correctly, in this context when we have data and schema unified, i.e, when data matches a schema, we are effectively instantiating a value out of that data, whose type is fully understood by the implementing language.

Btw, one more ask related to **schema-matching**, is this something similar to erlang/haskell pattern matching ? If so, both shape of the schema and type of its field considered ? If there is a document explaining the schema matching exist, we can link it here.

Reply:
Ah, I kind of like that nomenclature when you explain it like that.

It has somewhat limited ability to be communicated by implication, I'm afraid, because some of those words are so synonymous colloquially that a reader might not reliably be able to detect from usage that the words are meant to be load-bearing of different meaning.  But when it's laid out clearly it's very nice.

I'll think more on this in the future.  :)

Reply:
Oh!  Regarding schema-matching: there's some documentation available in this in the vicinity of https://specs.ipld.io/schemas/migration.html#version-detection .  It's likely that documentation is due for some expansions too, though!  And it's probably going to feel somewhat similar to pattern matching in various programing languages, yep.

It's all structural -- there really isn't any types associated with data until _after_ this matching happens, so structural information is all it's got to work with.

(I guess it would be possible to perform schema matching *again* on some other data that had already gone through one schema... but it would be a little odd and I wouldn't expect people to often find any reason to do this in practice.  And the main reason it happens to work is that even schema-processed data still implements the same interfaces as the raw Data Model... so, it would only be reasonable that even if someone did try to do a schema match on other already-schema'd data, it should still treat it as the data alone, for consistency.)